src/server/debug/server/server.go

```diff
@@ -430,15 +431,15 @@ func (s *debugServer) collectInputRepos(tw *tar.Writer, pachClient *client.APICl
 		if _, err := pachClient.InspectPipeline(repoInfo.Repo.Name, true); err != nil {
 			if errutil.IsNotFoundError(err) {
 				repoPrefix := join("source-repos", repoInfo.Repo.Name)
-				return s.collectCommits(tw, pachClient, repoInfo.Repo, limit, repoPrefix)
+				return s.collectCommits(ctx, tw, pachClient, repoInfo.Repo, limit, repoPrefix)
 			}
 			return err
 		}
 	}
 	return nil
 }
 
-func (s *debugServer) collectCommits(tw *tar.Writer, pachClient *client.APIClient, repo *pfs.Repo, limit int64, prefix ...string) error {
+func (s *debugServer) collectCommits(rctx context.Context, tw *tar.Writer, pachClient *client.APIClient, repo *pfs.Repo, limit int64, prefix ...string) error {
```

Comment:
`rctx`?

Reply:
root context, so ctx doesn't shadow itself when we create the cancelable context.  I think if you call the context you don't want to ever use "ctx", it's easy to accidentally use it and be confused as to why cancelation doesn't work; that's why I didn't call this one "ctx" and the other one "cctx".